### DISCUSSION TITLE
[Feat] [@mantine/form] Add setInitialValues function, allow modify initialValues after useForm was called

### DISCUSSION THREAD

**Discussion by Ryan2128** [mantinedev/mantine#4864] [Feature requests]
## Description

When use a form to process two operations, such as create and update, initialValues can only be set once, after `useForm` has been called, there is no way to modify initialValues.

## Why need this
The reason why need to modify initialValues is when use a form to process two operations, such as create and update, initialValues can only be set once, after `useForm` has been called, there is no way to modify initialValues.

This leads to switching to the update operation after the form is initialized, and the initialValues of the update operation form cannot be set.

Therefore, I had to use two components, and each component used useForm once to implement different initial values. This caused a lot of repeated code, so I wanted to implement this function to avoid repeated code.

## Others

Just like this PR[#4556] present. 

I found that this PR is now closed. The reason for closure is that it may have been implemented in v7.0.0, or the PR is no longer relevant to v7.0.0, or it is due to follow the contribution workflow. After I checked the source code of the form module of v7.0.0, I found that this function has not been implemented, so I guessed that the last reason prevented the PR from being merged, so I opened this Feat issue.

@rtivital 

If the new feature is approved, I can continue to complete the v7.0.0 implementation of the feature.

**Comment by rtivital** [comment]
I plan to add this feature, but implement this differently than the PR

**Reply by Ryan2128** [reply]
Thanks, looking forward to this feature being added.
